test/controllers/order_items_controller_test.rb

```diff
@@ -0,0 +1,65 @@
+require "test_helper"
+
+describe OrderItemsController do
+
+  let(:order) {
+    Order.create!(
+        email: 'someone@something.com',
+        address: 'some address',
+        credit_name: 'Someone',
+        credit_expire: '11/22',
+        security_code: 'secret code',
+        zip: 111111,
+        credit_card_name: "1234"
+    )
+  }
+
+  let(:order_item){
+    OrderItem.create!(
+        quantity: 3,
+        product_id: product.id,
+        order_id: order.id,
+        shipped: false
+    )
+  }
+
+  describe "Create" do
+    it "redirect when there is inavlid product ID" do
+      post product_order_items_path(-1)
+      expect(flash[:error]).must_equal "Invalid product"
+      must_respond_with :redirect
+      must_redirect_to root_path
+
+    end
+
+    it "can add product to the shopping cart and redirect" do
+      skip
+      post product_order_items_path(products(:product_one))
+      must_respond_with :redirect
+      must_redirect_to cart_path
+    end
+
+  end
+
+  describe "Destroy" do # to do later after finish
+
+    it "delete a product from the shopping cart and redirect" do
+      skip
+      post product_order_items_path(products(:product_one))
+      id = Product.find_by_id(products(:product_one).id)
+      expect {
+        delete product_order_item_path(id)}.must_change 'OrderItem.count', -1
```

Comment:
Your form requires the quantity you're adding to the product, and you need both the product and order ids.

```suggestion
      post product_order_items_path(products(:product_one).id), params: { order_item: { quantity: 1 } }
      product_id = Product.find_by_id(products(:product_one).id).id
      id = OrderItem.find_by(product_id: product_id, order_id: session[:cart_id]).id
      expect {
        delete product_order_item_path(product_id, id)
      }.must_change 'OrderItem.count', -1
```